tests/unit/streamalert/classifier/test_parsers_json.py

```diff
@@ -1091,3 +1091,32 @@ def test_extract_via_json_regex_key_no_key(self):
         parser = JSONParser(options)
         result = parser._extract_via_json_regex_key(record_data)
         assert_equal(result, False)
+
+    def test_parse_record_not_dict_mismatch(self):
+        """JSONParser - Parse record not in dict type and doesn't match schema"""
+        options = {
+            'schema': {
+                'key': 'string'
+            },
+            'parser': 'json'
+        }
+        record_data = "[{\"key\": \"value\"}]"
+
+        parser = JSONParser(options)
+        assert_equal(parser.parse(record_data), False)
+
+    def test_parse_record_not_dict_matched(self):
+        """JSONParser - Parse record not in dict type but match the schema"""
+        options = {
+            'schema': {
+                'key': 'string'
+            },
+            'parser': 'json',
+            'configuration': {
+                'json_path': "[*]"
+            }
+        }
+        record_data = "[{\"key\": \"value\"}]"
+
+        parser = JSONParser(options)
+        assert_equal(parser.parse(record_data), True)
```

Comment:
can we also add a check for the actual value of the parsed records here? I want to ensure this is for sure doing what we'd expect.

see: https://github.com/airbnb/streamalert/blob/b6c7960e937e1657c0bca15f5c4068cd66418504/tests/unit/streamalert/classifier/test_parsers_json.py#L88-L89